### PR TITLE
Issue #3655 - Cookie generation now complies with RFC6265 spaces

### PR DESCRIPTION
**Pull request by joakime**
Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

File: jetty-server/src/main/java/org/eclipse/jetty/server/CookieCutter.java [hunk 1]
```diff
@@ -17,25 +17,31 @@
...
- * If the added fields are identical to those last added (as strings), then the 
- * cookies are not re parsed.
- *
+/**
+ * Cookie parser
+ * <p>
+ *     Optimized stateful {@code Cookie} header parser.
+ *     Does not support {@code Set-Cookie} header parsing.
+ * </p>
+ * <p>>
```
Comment (sbordet) [inline jetty-server/src/main/java/org/eclipse/jetty/server/CookieCutter.java]: Extra `>`?

**Comment by gregw** [comment]
@joakime I created a fix2 branch from yours where I have refactored the replace cookie mechanism to either used a stored jetty HttpCookie, or to use the java.net.HttpCookie class to parse one that has been directly set:

https://github.com/eclipse/jetty.project/compare/jetty-9.4.x-3655-setcookie-rfc6265-fix...jetty-9.4.x-3655-setcookie-rfc6265-fix2

I think this is a better direction... but perhaps jetty-10 only?

**Comment by joakime** [comment]
@gregw nah, I think the fixes you did are generally useful, even in Jetty 9.4.x
I'd like to merge both this PR, and your branch into `jetty-9.4.x` today.

**Comment by joakime** [comment]
Closing in favor of @gregw 's changes in PR #3664

**Pull request closed without merging**